presto-orc/src/main/java/com/facebook/presto/orc/OrcWriteValidation.java

```diff
@@ -265,17 +271,19 @@ public void validateRowGroupStatistics(
         }
 
         RowGroupStatistics expectedRowGroup = rowGroups.get(rowGroupIndex);
-        RowGroupStatistics actualRowGroup = new RowGroupStatistics(IntStream.range(1, actual.size()).boxed().collect(toImmutableMap(identity(), actual::get)));
+        RowGroupStatistics actualRowGroup = new RowGroupStatistics(false, IntStream.range(1, actual.size()).boxed().collect(toImmutableMap(identity(), actual::get)));
 
-        Map<Integer, ColumnStatistics> expectedByColumnIndex = expectedRowGroup.getColumnStatistics();
+        if (!expectedRowGroup.isLowMemoryMode()) {
+            Map<Integer, ColumnStatistics> expectedByColumnIndex = expectedRowGroup.getColumnStatistics();
 
-        // new writer does not write row group stats for column zero (table row column)
-        List<ColumnStatistics> expected = IntStream.range(1, actual.size())
-                .mapToObj(expectedByColumnIndex::get)
-                .collect(toImmutableList());
-        actual = actual.subList(1, actual.size());
+            // new writer does not write row group stats for column zero (table row column)
+            List<ColumnStatistics> expected = IntStream.range(1, actual.size())
+                    .mapToObj(expectedByColumnIndex::get)
+                    .collect(toImmutableList());
+            actual = actual.subList(1, actual.size());
 
-        validateColumnStatisticsEquivalent(orcDataSourceId, "Row group " + rowGroupIndex + " in stripe at offset " + stripeOffset, actual, expected);
+            validateColumnStatisticsEquivalent(orcDataSourceId, "Row group " + rowGroupIndex + " in stripe at offset " + stripeOffset, actual, expected);
+        }
 
         if (expectedRowGroup.getHash() != actualRowGroup.getHash()) {
```

Comment:
Wonder if we wanna put this check only in `isLowMemoryMode` just in case we may miss something and give false alert. For example (the metadata reader strips out string stats when min/max = null causing different hash result). So it won't affect production runs